src/index.js

```diff
@@ -36,16 +44,29 @@ module.exports = class RpcEngine extends SafeEventEmitter {
 
   async _handleBatch (reqs, cb) {
 
+    let batchRes
+    let error = null
+
     // The order here is important
     try {
-      const batchRes = await Promise.all( // 2. Wait for all requests to finish
+      // 3a. Store batch response
+      batchRes = await Promise.all( // 2. Wait for all requests to finish
         // 1. Begin executing each request in the order received
         reqs.map(this._promiseHandle.bind(this)),
       )
-      return cb(null, batchRes) // 3a. Return batch response
     } catch (err) {
-      return cb(err) // 3b. Some kind of fatal error; all requests are lost
+      // 3b. Some kind of fatal error, all requests are lost
+      error = err
+    }
+
+    if (cb) {
```

Comment:
It might be cleaner/shorter for this callback-wrapper to live in `handle` instead (on line 34) and leave this function as a regular `async` function.

Reply:
Ahh, I was hoping you'd be able to see how to make it less messy 😄 
I'd been staring at it for too long, haha.

Done in: 28bd82e